### DISCUSSION TITLE
Listing all files with a name pattern, excluding searching some directories by pattern

### DISCUSSION THREAD

**Discussion by MidGe48** [nushell/nushell#12805] [Q&A]
I am just looking at nushell and would like some help.

I am running BTRFS opn my computer. I would like to find all occurrences of, say, "nushell" on my computer excluding the directories named ".snapshot*" which are btrfs past snapshots on my system and thus contain many repeats of the file name I am searching for.

Is it possible to do this in nushell. All my attempts have failed, but as I said, I am just looking at nushell potential.

Thanks for any help.

**Comment by devyn** [comment]
Yes, our `glob` command is quite sophisticated, have a look at `glob --help`

This might work:

```nushell
glob **/*nushell*/** --exclude [**/.snapshot*/**]
```

**Reply by MidGe48** [reply]
Thank you for your prompt reply, unfortunately that is not working for me. It returns an empty list when, in my zsh shell, -- find / -not -path "*/\*snapshot\*/*" -name "nushell" -- returns the correct info.
The glob --help did not enlighten me but that may be because of missing basic understanding since I only started looking at this today.
My interest was based on the fact that it is written in Rust.

Thanks again
